src/vuex/store.js

```diff
@@ -37,46 +39,99 @@ const store = new Vuex.Store({
         creatures: [],
       };
 
+      function getDir(x, y) {
+        let dir = [];
+        if (y < 0) dir.push("north");
+        else if (y > 0) dir.push("south");
+        if (x < 0) dir.push("west");
+        else if (x > 0) dir.push("east");
+        if (!x && !y) dir.push("here");
+        return dir.join(" ");
+      }
+
       Object.keys(state.creatures).forEach(key => {
         let pos = key.split(",");
         let x = parseInt(pos[0]) - state.player.pos.x;
         let y = parseInt(pos[1]) - state.player.pos.y;
 
         if (Math.abs(x) <= radius && Math.abs(y) <= radius && state.creatures[key].length > 0) {
           state.creatures[key].forEach(creature => {
-            let dir = [];
-            if (y < 0) dir.push("north");
-            else if (y > 0) dir.push("south");
-            if (x < 0) dir.push("west");
-            else if (x > 0) dir.push("east");
-            if (!x && !y) dir.push("here");
-            creature.dir = dir.join(" ");
+            creature.dir = getDir(x, y);
             creature.dist = Math.abs(x) + Math.abs(y);
             creature.loc = [x, y];
             surr.creatures.push(creature)
           });
         }
       });
+      
+      surr.creatures.sort((a, b) => {
+        return a.dist - b.dist === 0 ? b.level - a.level : a.dist - b.dist;
+      });
 
+      let items = {
+        total: 0,
+        stacked: {},
+      };
       for (let y = -radius; y <= radius; y++) {
         for (let x = -radius; x <= radius; x++) {
           let tile = state.world.getTile(state.player.pos.x + x, state.player.pos.y + y);
           if (tile.items.length) {
-            surr.items.push(...tile.items);
+            items.total += tile.items.length;
+            tile.items.forEach(item => {
```

Comment:
This whole forEach loop feels like unbelievably hacky code. It works, but I don't want it too. I'd appreciate if you could give me some pointers as to how I could better structure this data for using in the surroundings pane @Alpvax

Reply:
This is really hard to follow, I'm not even sure what structure you have here, or what you are after.

Reply:
I agree. I just kept adding bits until it worked, hence the spaghetti.
What I want at the end is an object
```{
stacked: {
  bone: {
    count: 3,
    expanded: {
      -1,1,bone: {
        count: 3,
        dir: "south west",
        loc: {
          5,8: {}
        },
        name: "bone",
        plural: "bones",
        totalValue: 75,
      },
    },
    locations: {5,8: {…}},
    name: "bone",
    plural: "bones",
  dragon blood: {
    count: 1,
    expanded: {2,-2,dragon blood: {…}},
    locations: {8,5: {…}},
    name: "dragon blood",
    plural: "dragon blood",
  },
  dragon claws: {name: "dragon claws", plural: "dragon claws", count: 7, locations: {…}, expanded: {…}},
  dragon hide: {name: "dragon hide", plural: "dragon hides", count: 2, locations: {…}, expanded: {…}},
  dragon meat: {name: "dragon meat", plural: "dragon meat", count: 2, locations: {…}, expanded: {…}},
  dragon scales: {name: "dragon scales", plural: "dragon scales", count: 6, locations: {…}, expanded: {…}},
  dragon skull: {name: "dragon skull", plural: "dragon skulls", count: 1, locations: {…}, expanded: {…}},
  dragon teeth: {name: "dragon teeth", plural: "dragon teeth", count: 6, locations: {…}, expanded: {…}},
  skull: {name: "skull", plural: "skulls", count: 1, locations: {…}, expanded: {…}},
},
total: 29,
}
```
That is, I want the total number of items in the surroundings, given by `total`. I then want a list of each of the items in the surroundings, which I store in an object called `stacked` with the item names as the keys so I can stack them (I check if that item name already exists in my `stacked` object).
Then, for each `stacked` object, I store the item's `name`, `plural` name, the number in the surroundings (`count`), the `locations` as an array so I can pass it into the highlight function in the store, and an `expanded` object.
The `expanded` object expands all of the objects with that name in the surroundings into a list of each item separated per tile, so the keys I use for that are the relative location from the player and the name, e.g. `-1,1,bone`, which JS stringifies from a 3-length array for me. As I type this, I realise that because of how I'm nesting these objects, I don't need the item name in the key, but it doesn't really matter.

Reply:
I think using a custom type would be better here, the `expanded` seems to be a lot of duplication.

Reply:
What do you mean by that? Make it a type instead of an object?

Reply:
What doesn't work. Your suggestion or my code?

Reply:
My suggestion. Vue never updates